File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/busy/FinishedCallback.java
package com.cedarsoft.commons.javafx.busy;

/**
 * Is called when a system command has been finished
 */
public interface FinishedCallback {
  FinishedCallback NONE = new FinishedCallback() {
    @Override
    public void finished() {
    }

    @Override
    public void timedOut() {
    }
  };

  /**
   * Is called when a task has been finished
   */
  void finished();

  /**
   * Is called when a task has timed out
   */
  void timedOut();
}

File: commons/javafx/src/main/kotlin/it/neckar/open/javafx/busy/FinishedCallback.kt
package it.neckar.open.javafx.busy

/**
 * Is called when a system command has been finished
 */
interface FinishedCallback {
  /**
   * Is called when a task has been finished
   */
  fun finished()

  /**
   * Is called when a task has timed out
   */
  fun timedOut()

  companion object {
    @JvmStatic
    val None: FinishedCallback = object : FinishedCallback {
      override fun finished() {}
      override fun timedOut() {}
    }
  }
}
